### DISCUSSION TITLE
Environments naming rule

### DISCUSSION THREAD

**Discussion by taozuhong** [msys2/MINGW-packages#17071] [General]
![image](https://user-images.githubusercontent.com/13451009/236590140-db721d2c-96ad-46d4-99b6-367dd92c64ed.png)

Icon    |Name      | Prefix    | Toolchain    | Architecture | C Library | C++ Library
------- |--------- | --------- | ------------ | ------------ |  -------- | ------------
<img src="https://www.msys2.org/docs/msys.png" height="24" width="24">| MSYS    | /usr     | gcc       | x86_64       | Cygwin    | libstdc++
<img src="https://www.msys2.org/docs/msys.png" height="24" width="24">| MSYSX64    | /usrx64     | gcc       | x86_64       | Cygwin    | libstdc++
<img src="https://www.msys2.org/docs/msys.png" height="24" width="24">| MSYSA64 | /usra64  | gcc       | aarch64       | Cygwin    | libstdc++
<img src="https://www.msys2.org/docs/msys.png" height="24" width="24">| MSYSR64 | /usrr64  | gcc       | riscv64       | Cygwin    | libstdc++
<img src="https://www.msys2.org/docs/ucrt64.png" height="24" width="24">| UCRT64  | /ucrt64  | gcc | x86_64 | ucrt | libstdc++
<img src="https://www.msys2.org/docs/ucrt64.png" height="24" width="24">| UCRTX64  | /ucrtx64  | gcc | x86_64 | ucrt | libstdc++
<img src="https://www.msys2.org/docs/ucrt64.png" height="24" width="24">| UCRTA64 | /ucrta64 | gcc | aarch64 | ucrt | libstdc++
<img src="https://www.msys2.org/docs/ucrt64.png" height="24" width="24">| UCRTR64 | /ucrtr64 | gcc | riscv64 | ucrt | libstdc++
<img src="https://www.msys2.org/docs/clang64.png" height="24" width="24">| CLANG64 | /clang64 | llvm | x86_64 | ucrt | libc++
<img src="https://www.msys2.org/docs/clang64.png" height="24" width="24">| CLANGX64 | /clangx64 | llvm | x86_64 | ucrt | libc++
<img src="https://www.msys2.org/docs/clang64.png" height="24" width="24">| CLANGA64 | /clanga64 | llvm | aarch64 | ucrt | libc++
<img src="https://www.msys2.org/docs/clang64.png" height="24" width="24">| CLANGR64 | /clangr64 | llvm | riscv64 | ucrt | libc++
<img src="https://www.msys2.org/docs/mingw64.png" height="24" width="24">| MINGW64 | /mingw64 | gcc | x86_64 | msvcrt | libstdc++
<img src="https://www.msys2.org/docs/mingw64.png" height="24" width="24">| MINGWX64 | /mingwx64 | gcc | x86_64 | msvcrt | libstdc++
<img src="https://www.msys2.org/docs/mingw64.png" height="24" width="24">| MINGWA64 | /mingwa64 | gcc | aarch64 | msvcrt | libstdc++
<img src="https://www.msys2.org/docs/mingw64.png" height="24" width="24">| MINGWR64 | /mingwr64 | gcc | riscv64| msvcrt | libstdc++

**Comment by rimrul** [comment]
> Architecture: riscv64

Is there even a Windows port for that?

> CLANGA64

We shouldn't rename this.
Edit: I originally wrote UCRTA64, but meant CLANGA64

> MINGWA64

I don't think there is an ARM64 msvcrt.

> MINGWX64, CLANGX64, UCRTX64, MSYSX64

Why should all of these exist twice?

> UCRTA64, MSYSA64

I don't think there's currently gcc tooling to attempt to build these.

>MSYSA64

The prefix would probably need to stay `/usr`, just like i686 MSYS used to have `/usr`.

**Reply by taozuhong** [reply]
`MINGWX64, CLANGX64, UCRTX64, MSYSX64`
Just show them here for uniform and compatibility.

**Comment by taozuhong** [comment]
@lazka @Biswa96 
Could you pls take a look at it?

**Reply by lazka** [reply]
I don't understand what you are trying to say with this.

**Reply by taozuhong** [reply]
Just share some thoughts about environment name, because there more ENV will be added in the future.

**Comment by wszqkzqk** [comment]
> MINGWA64

`msvcrt.dll` is deprecated and there is no reason to port `MINGW64` to other platforms.

Also, the gcc port of windows for arm seems not ready yet. I think it's not the time for msys2 to add these things.

> RISC-V

And no one knows windows for risc-v so it's meaningless to discuss here...

**More importantly, we should not break compatibility just for personal preference.**

**Comment by Blackhex** [comment]
@taozuhong Note, that there is `MINGWARM64` instead of `MINGWA64` appearing in the code base and dicussions (e.g. https://github.com/search?q=repo%3Amsys2%2FMINGW-packages%20mingwarm64&type=code)

@wszqkzqk Can you please explain a bit why there is no reason to port GNU toolchain to aarch64?
> msvcrt.dll is deprecated and there is no reason to port MINGW64 to other platforms.

**Reply by rimrul** [reply]
> Can you please explain a bit why there is no reason to port GNU toolchain to aarch64?

>> msvcrt.dll is deprecated and there is no reason to port MINGW64 to other platforms.

That comment was probably about a GCC toolchain that links against MSVCRT (which AFAIK doesn't exist for aarch64), not about GCC toolchains in general (see the environments named UCRT... in the original post)

**Reply by Blackhex** [reply]
@rimrul Thank you for the reaction. Yes, that is my interest. So far I tought msvcrt is the "default" and ucrt just an "advanced option" from MSYS2 perspective...
